### DISCUSSION TITLE
Hyperlink Button & Windows.System.LaunchUriAsync() not working on iOS

### DISCUSSION THREAD

**Discussion by trydalch** [unoplatform/uno#18562] [Q&A] [answered]
We're trying to add links to our privacy policy and EULA to meet apple's app store requirements. Both the hyperlinks and the buttons work when running on windows, but do not on iOS (see snippets below). 
```xml
      <Grid HorizontalAlignment="Stretch" RowSpacing="20" >
        <Grid.ColumnDefinitions>
          <ColumnDefinition Width="*" />
          <ColumnDefinition Width="*" />
        </Grid.ColumnDefinitions>

        <Grid.RowDefinitions>
          <RowDefinition Height="Auto" />
          <RowDefinition Height="Auto" />
        </Grid.RowDefinitions>


        <HyperlinkButton Grid.Column="0"
                         HorizontalAlignment="Center"
                         Content="Privacy Policy"
                         NavigateUri="https://www.wereallrelated.com/privacy_policy.html" />

        <HyperlinkButton Grid.Column="1"
                         HorizontalAlignment="Center"
                         Content="Terms of Use"
                         NavigateUri="https://www.apple.com/legal/internet-services/itunes/dev/stdeula" />

        <Button Grid.Column="0"
                Grid.Row="1"
                Style="{StaticResource TextButtonStyle}"
                Content="Privacy Policy"
                Tag="https://www.wereallrelated.com/privacy_policy.html"
                HorizontalAlignment="Center"
                Click="On_Button_Hyperlink_Click" />

        <Button Grid.Column="1"
                Grid.Row="1"
                Style="{StaticResource TextButtonStyle}"
                Content="Terms of Use"
                Tag="https://www.apple.com/legal/internet-services/itunes/dev/stdeula"
                HorizontalAlignment="Center"
                Click="On_Button_Hyperlink_Click" />
        
      </Grid>
```

```csharp
private async void On_Button_Hyperlink_Click(object sender, RoutedEventArgs e)
{
	// Launch the url from the button tag
	if (sender is Button button)
	{
		if (button.Tag is string url)
		{
			if (Uri.TryCreate(url, UriKind.RelativeOrAbsolute, out Uri uri))
			{
				var result = await Windows.System.Launcher.LaunchUriAsync(uri);
			}

		}
	}
}
```

I put a HyperlinkButton in the project below to test. It would not work for me when running in debug on iOS in Visual Studio. 
[SimpleCalculator.zip](https://github.com/user-attachments/files/17497996/SimpleCalculator.zip)

**Comment by jeromelaban** [comment]
Thanks for the report. Could you indicate which version of Uno you're using? Also, do you see any exception in the app's logs?

**Reply by MartinZikmund** [reply]
@trydalch We tested the attached repro on iOS device and it worked successfully. Do you see some log messages that could help?

**Reply by Asfiroth** [reply]
I have a similar scenario and facing the same issue, whenever user clicks the button and `LaunchUriAsync` is called this is logged :`[Application] BUG IN CLIENT OF UIKIT: The caller of UIApplication.openURL(_:) needs to migrate to the non-deprecated UIApplication.open(_:options:completionHandler:). Force returning false (NO).`

I'm using Uno Sdk 5.4.10, not sure if this has anything to do with debugging in an iOS simulator (iOS 18, XCode 16)
@jeromelaban @MartinZikmund

**Reply by spouliot** [reply]
The UIKit [openURL](https://developer.apple.com/documentation/uikit/uiapplication/1622961-openurl) API, being used inside Uno, has been deprecated for a while inside iOS.

On recent versions of iOS Apple has gone a step further and now force apps to use the newer API. This is the error you're seeing inside your logs.

> We tested the attached repro on iOS device and it worked successfully. 

It was probably running an older version of iOS.

**Reply by MartinZikmund** [reply]
Indeed, it is likely we ran against iOS 17.5 here

**Reply by MartinZikmund** [reply, marked as answer]
@trydalch Turned this into an issue, follow the progress there please https://github.com/unoplatform/uno/issues/18621
